src/engines/ws.ts

```diff
@@ -70,40 +70,48 @@ export class WebsocketEngine extends AbstractEngine {
 			});
 
 			socket.addEventListener("message", async ({ data }) => {
-				const decoded = this.decodeCbor(
-					data instanceof Blob
-						? await data.arrayBuffer()
-						: data.buffer.slice(
-								data.byteOffset,
-								data.byteOffset + data.byteLength,
-							),
-				);
-
-				if (
-					typeof decoded === "object" &&
-					!Array.isArray(decoded) &&
-					decoded != null
-				) {
-					this.handleRpcResponse(decoded);
-				} else {
-					this.setStatus(
-						ConnectionStatus.Error,
-						new UnexpectedServerResponse(decoded),
+				try {
+					const decoded = this.decodeCbor(
+						data instanceof Blob
+							? await data.arrayBuffer()
+							: data.buffer.slice(
+									data.byteOffset,
+									data.byteOffset + data.byteLength,
+								),
 					);
+
+					if (
+						typeof decoded === "object" &&
+						decoded != null &&
+						Object.getPrototypeOf(decoded) === Object.prototype
```

Comment:
What about null prototype that is often used for better security against prototype pollution

example of how to set it:
```js
{__proto__: null}
```

this is just one example, you can also use `Object.create(null)`

Reply:
What exactly are you aiming at?